Core/WumpusCore/GameLocations/GameLocations.cs

```diff
@@ -25,22 +25,34 @@ public enum RoomTypes
         /// <summary>
         /// The random object from the controller class.
         /// </summary>
-        Random random;
+        private Random random;
 
         public GameLocations()
         {
             rooms = new List<Room>();
             random = Controller.Controller.Random;
         }
 
+        /// <summary>
+        /// Initializes the <c>rooms</c> list with rooms of <c>RoomTypes</c> type <c>Flats</c>.
+        /// </summary>
+        /// <param name="numRooms">The total amount of rooms.</param>
+        public void InitRooms(ushort numRooms)
+        {
+            for (ushort i  = 0; i < numRooms; i++)
+            {
+                AddRoom(RoomTypes.Flats, (ushort)(i+1));
+            }
+        }
+
         /// <summary>
         /// Gets a random empty room.
         /// </summary>
-        /// <returns>A random empty room.</returns>
-        public int GetEmptyRoom()
+        /// <returns>A random room of <c>RoomTypes</c> type <c>Flats</c> from the <c>rooms</c> list.</returns>
+        public ushort GetEmptyRoom()
```

Comment:
I know it isn't my place to review this but you could be much more efficient if you grabbed random rooms until you found a RoomTypes.Flats. Also why is it RoomTypes and not just RoomType?

Reply:
I was planning on adding an exception if there were no empty rooms left, which would be a little hard with the method you suggested.
I called the variable RoomTypes because it stores all the possible RoomTypes - I was thinking of it more as variable than a type. Since I'm not very experienced with enums, are they usually named as types rather than variables? Would that also mean that the Directions enum in topology is also improperly named?

Reply:
If you really want RoomTypes to be renamed to RoomType, just lmk but for now I'll just keep it as is.